addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/AuthenticationDiagnostics.java

```diff
@@ -199,9 +201,55 @@ public void recordStep(WebDriver wd, String description, WebElement element) {
             }
         }
 
+        if (wd instanceof JavascriptExecutor je) {
+            processStorage(
+                    je,
+                    """
+                    const data = [];
+                    for (let i = 0; i < window.localStorage.length; i++) {
+                        data.push({"key": window.localStorage.key(i), "value": window.localStorage.getItem(window.localStorage.key(i))});
+                    }
+                    return data;
+                    """,
+                    DiagnosticBrowserStorageItem.Type.LOCAL);
+
+            processStorage(
+                    je,
+                    """
+                    const data = [];
+                    for (let i = 0; i < sessionStorage.length; i++) {
+                        data.push({"key": sessionStorage.key(i), "value": sessionStorage.getItem(sessionStorage.key(i))});
+                    }
+                    return data;
+                    """,
```

Comment:
Use a single or constant string and a variable for the storage type/location and insert it via format?

Reply:
Not 100% sure this is what you meant but it addresses the duplication.

Reply:
Yup, thanks!